ipa-core/src/protocol/ipa_prf/malicious_security/verifier.rs

```diff
@@ -145,14 +132,15 @@ where
 
     // generate & evaluate recursive streams
     // to compute last array
-    let mut stream: Pin<Box<dyn Stream<Item = F> + Send>> = Box::pin(
-        recurse_u_or_v::<_, _, λ_FIRST>(u_or_v_stream, &table_first),
-    );
+    let mut iterator: Box<dyn Iterator<Item = F>> = Box::new(recurse_u_or_v::<_, _, λ_FIRST>(
```

Comment:
Do you need to include this type? Can you get away with just:
```
let mut iterator = Box::new(recurse_u_or_v::<_, _, λ_FIRST>(
```

Reply:
unfortunately not. The dyn trait allows any iterator. When I do not specify the type, the compiler thinks that it is a Box around the return type of recurse_u_or_v after the first recursion